core/src/main/java/jenkins/model/ArtifactManager.java

```diff
@@ -60,11 +60,17 @@
      * @param launcher a launcher to use if external processes need to be forked
      * @param listener a way to print messages about progress or problems
      * @param artifacts map from paths in the archive area to paths relative to {@code workspace} (all paths {@code /}-separated)
+     * @param preserveExecute if true, sets the execute bit on the target if the execute bit is set on the source
+     * @param preserveMtime if true, sets the modification time of the target to match the modification time of the source
      * @throws IOException if transfer or copying failed in any way
      * @throws InterruptedException if transfer was interrupted
      * @see ArtifactArchiver#perform(Run, FilePath, Launcher, TaskListener)
      */
-    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener, Map<String,String> artifacts) throws IOException, InterruptedException;
+    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener,
```

Comment:
I see. Then there are two options. 1) We can change FilePath.recursiveCopyTo() to always preserve the attributes, or 2) we can leave option to preserve permissions in FilePath.recursiveCopyTo and call the recursiveCopyTo with hardcoded parameters for attribute preservation.

I chose the configurable option because I did not know if any callers of these interfaces requite the attributes to be dropped. But if intended behavior of FilePath.recursiveCopyTo is that attributes should be preserved, then the patch can be reduced to two lines in FilePath.java.